From 96b4045c746c245a2834eeb6e280f033094ed9cb Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 30 Jul 2013 12:39:03 +0300
Subject: [PORT FROM MAIN] Telephony: Call onHangupLocal on rejectCall request

BZ: 31496

Upon call disconnected state, disconnect cause
is checked for determining whether the hangup is
local or missed call. Currently, cause is not
set on user rejecting the call.

Based on the cause, UI message "incoming" or "missed"
call will be shown. So set the cause when the
user rejects the call.

Change-Id: Id403eabf611a81cd553f9e98ffa5ff0fac562aa2
Orig-Change-Id: I566b65d6bf6b1fe6187c4a72f628f199795cdc61
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../internal/telephony/gsm/GsmCallTracker.java     |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
index 07b3c6e..583b28c 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
@@ -283,6 +283,8 @@ public final class GsmCallTracker extends CallTracker {
         // so if the phone isn't ringing, this could hang up held
         if (mRingingCall.getState().isRinging()) {
             mCi.rejectCall(obtainCompleteMessage());
+            mRingingCall.onHangupLocal();
+            mPhone.notifyPreciseCallStateChanged();
         } else {
             throw new CallStateException("phone not ringing");
         }
-- 
1.7.4.1

